specification/data-semantic-conventions.md

```diff
@@ -57,6 +58,33 @@ and has a lower cardinality than the URI path or route, this value MUST be used
 | `http.route` | The matched route. E.g. `"/users/:userID?"`. | No |
 | `http.status_code` | [HTTP response status code](https://tools.ietf.org/html/rfc7231). E.g. `200` (integer) | No |
 | `http.status_text` | [HTTP reason phrase](https://www.ietf.org/rfc/rfc2616.txt). E.g. `"OK"` | No |
+| `http.app` | An identifier for the whole HTTP application. E.g. Flask app name, `spring.application.name`, etc. | No |
+| `http.app_root` |The path prefix of the URL that identifies this `http.app`. If multiple roots exist, the one that was matched in the current URL should be used. | No |
+| `http.client_ip` | The IP address of the original client behind all proxies, if known. For syntax, see `peer.ip`. | No |
+
+It is recommended to also use the general [network attributes][], especially `peer.ip` and `host.name` as well as the `code.ns` and `code.func` [code attributes][] to name the logical handler method (`code.ns` + `code.func` will have a lower cardinality than `http.route`).
+
+As an example, if a browser request for `https://example.com/webshop/articles/4` is invoked, we may have:
+
+Span name: `/webshop/articles/:article_id` (`app_root` + `route`).
+
+|   Attribute name   |                      Value                      |
+| :----------------- | :---------------------------------------------- |
+| `component`        | `"http"`                                        |
+| `http.method`      | `"GET"`                                         |
+| `http.url`         | `"https://example.com/webshop/articles/4"`      |
```

Comment:
https seems to be wrong here.
Please note also that webservers usually get only `GET /webshop/articles/4 HTTP/1.1` from HTTP protocol.
To get the complete url used by client various headers like `host`, `forwarded`,... can be used but not sure if there is a clear specification somewhere on how this should be done by an opentelemetry sensor.

Reply:
Why do you think that https is wrong?

Very valid point with the URL (though usually host header and the used protocol are enough to reconstruct it). I plan to push an update that allows/recommends setting either: URL or Host-Header + scheme + the whole path part OR (if host header is missing) instead of it some configured server name + port.

Reply:
> Why do you think that https is wrong?

because this is for http. or does this apply also for https and http2 (and maybe even for H3/Quic)?

Reply:
It applies to all kinds  (see the example in the current version that also uses https https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md#http-server). I'll add a note.

Reply:
I added a note and the `http.flavor` field.